### PR TITLE
Fix for some 3180 warnings

### PR DESCRIPTION
**Pull request by mpetruska**
This tiny PR contains the fix for issue #840.

### CODE REVIEW

File: src/fsharp/tast.fs [hunk 1]
```diff
@@ -35,12 +35,12 @@ open Microsoft.FSharp.Core.CompilerServices
 /// Unique name generator for stamps attached to lambdas and object expressions
 type Unique = int64
 //++GLOBAL MUTABLE STATE
-let newUnique = let mutable i = 0L in fun () -> System.Threading.Interlocked.Increment(&i)
+let newUnique = let i = ref 0L in fun () -> System.Threading.Interlocked.Increment(i)
```
Comment (enricosada) [inline src/fsharp/tast.fs]: Mutable is ok. Warning is only informational
I disabled this warning ( converted to error ) in the project with nowarn

Comment (mpetruska) [inline src/fsharp/tast.fs]: Thanks, my thoughts exactly. Not too experienced with the build system yet, so I'd rather modify code than build parameters.

**Comment by dsyme** [comment]
This can be merged.  I'm Ok with making all allocations explicit in the compiler.

**Comment by enricosada** [comment]
@KevinRansom can you check and merge this? the debug build doesnt compile atm

**Comment by dsyme** [comment]
I'll merge this (as I need the debug build in any case)

**Comment by enricosada** [comment]
really thx @dsyme
Was really annoying to cherry-pick and remove the fix, and really bad for contributors

**Pull request merged**